packages/ember-glimmer/lib/utils/custom-component-manager.ts

```diff
@@ -0,0 +1,39 @@
+import { ComponentManager } from '@glimmer/runtime';
+
+import { assert } from 'ember-debug';
+import { Owner, symbol } from 'ember-utils';
+
+import DefinitionState from '../component-managers/definition-state';
+import ComponentStateBucket from '../utils/curly-component-state-bucket';
+
+import { GLIMMER_CUSTOM_COMPONENT_MANAGER } from 'ember/features';
+
+export const COMPONENT_MANAGER = symbol('COMPONENT_MANAGER');
+
+export function componentManager(obj: any, managerId: String) {
+  if ('reopenClass' in obj) {
+    return obj.reopenClass({
+      [COMPONENT_MANAGER]: managerId
+    });
+  }
+
+  return class extends obj {
+    static get [COMPONENT_MANAGER]() {
+      return managerId;
+    }
+  };
```

Comment:
Do we want to do this here instead of just mutating the class? The two branches do different things

Reply:
Are you saying this should be

```
obj[COMPONENT_MANAGER] = managerId;
```

instead of returning a new, extended class? That works for me. @rwjblue ?

Reply:
Yeah, sounds good to me